crates/ghost_actor/src/ghost_actor.rs

```diff
@@ -224,35 +226,38 @@ impl<
 
     fn priv_process_incoming_request(
         &mut self,
+        span: Span,
         message: P,
         maybe_cb: Option<GhostResponseCb<'lt, X, P>>,
     ) -> GhostResult<()> {
         match maybe_cb {
             Some(cb) => {
-                let request_id = self.pending_callbacks.bookmark(Span::fixme(), cb)?;
+                let request_id = self
+                    .pending_callbacks
+                    .bookmark(span.child("bookmarking"), cb)?;
 
-                self.send.send((Some(request_id), message))?;
+                self.send.send((span, Some(request_id), message))?;
             }
-            None => self.send.send((None, message))?,
+            None => self.send.send((span, None, message))?,
         }
         Ok(())
     }
 
     fn priv_process_handle_requests(&mut self, user_data: &mut X) -> GhostResult<()> {
-        while let Ok((maybe_id, message)) = self.recv.try_recv() {
+        while let Ok((span, maybe_id, message)) = self.recv.try_recv() {
```

Comment:
In the manual test_protocol the handlers have Span::fixme()'s in them that would go away if we send this span in through the `trigger` at the end of this function.

Reply:
When we talked, I heard you say no need for span in trigger(), so I tried not to, but I can change that no problem.